### ISSUE TITLE
Pkgconfig (Windows): gtksourceview-5 fails to import gtk-4

### ISSUE THREAD

**Issue by wroyca** [build2/build2#245]
Attempting to import [libgtksourceview](https://gitlab.gnome.org/GNOME/gtksourceview) fails with:

```
unable to execute C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.35.32124\bin\Hostx64\x64\cl: the parameter is incorrect
```

```
libs =
import libs += libadwaita-1%lib{adwaita-1}
import libs += libgtksourceview-5%lib{gtksourceview-5}

# NOTE: libecho is not a package.
#
include ../libecho/
libs += ../libecho/lib{echo}

# Resource files are written in XML and contain a list of resources and their 
# corresponding properties. The glib-compile-resources utility reads these 
# resource files and converts them into a binary representation that can be 
# used by Echo at runtime.
#
./: c{gresource}: xml{*}
{{
  diag xml ($<[0])
  glib-compile-resources --generate-source $path($<[0]) --sourcedir $src_root/echo
}}

./: exe{echo}: {c hxx ixx txx cxx}{**} $libs $gresource testscript

switch $cxx.target.class, $cxx.id
{
  case 'windows', 'msvc'
  {
    cxx.coptions = /MDd /Zi
    cxx.loptions = /DEBUG /LIBPATH:C:/gnome/lib
  }
}

cxx.poptions =+ "-I$out_root" "-I$src_root"
```

## Logs:
[logs.txt](https://github.com/build2/build2/files/10272693/logs.txt)

## Build:
```
meson setup builddir -Dprefix=C:/gnome
meson install -C builddir
```

**Comment by wroyca** [comment]
This is apparently caused by:

```
Requires: gtk4 >=  4.6
```

in

```
prefix=C:/gnome
includedir=${prefix}/include
libdir=${prefix}/lib

Name: gtksourceview-5
Description: Source code editing Gtk+ widget.
Version: 5.7.0
Requires: glib-2.0 >=  2.72, gobject-2.0 >=  2.72, gio-2.0 >=  2.72, gtk4 >=  4.6
Requires.private: fribidi >=  0.19.7, libxml-2.0 >=  2.6, pangoft2, fontconfig, libpcre2-8 >=  10.21
Libs: -L${libdir} -lgtksourceview-5
Cflags: -I${includedir}/gtksourceview-5
```

which is odd considering it work fine in libadwaita:

```
prefix=C:/gnome
includedir=${prefix}/include
libdir=${prefix}/lib

Name: Adwaita
Description: Adwaita Mobile widgets
Version: 1.3.alpha
Requires: gtk4 >=  4.9.1
Requires.private: glib-2.0 >=  2.72.0, fribidi, gio-2.0 >=  2.72.0
Libs: -L${libdir} -ladwaita-1
Cflags: -I${includedir}/libadwaita-1
```

**Comment by boris-kolpackov** [comment]
In the log that you have attached you can see three command lines at the end, just before the errors. Can you try to execute one of them and see if it fails? If it does, try to figure out what part of the command line is causing this (I don't see anything obvious).

On a more general note, when filing bug reports that only affect Windows (like this one) try to investigate things a bit further than "something doesn't work, here is the log". We are already spending a disproportionate amount of time on Windows issues and the further you investigate things the better the chance that we will be able to look into them.

**Comment by wroyca** [comment]
> On a more general note, when filing bug reports that only affect Windows (like this one) try to investigate things a bit further than "something doesn't work, here is the log". We are already spending a disproportionate amount of time on Windows issues and the further you investigate things the better the chance that we will be able to look into them.

I am oh so painfully aware - This is something I need to work on and I've been putting in effort to improve. 

In all honesty, It may not be obvious, but I do actually investigate - Here for instance I went through the log with MSDN documentation trying to spot something faulty and like you, not figuring out anything out of the ordinary ( tunnel vision? ) - That said, I did end up figuring out the "cause": `Requires: gtk4 >= 4.6` and managed to workaround the issue by tweaking manually the pkgconfig.

From this point, I figured out that any further investigation was to actually step-in and debug build2 internal itself, I recognize that this is not much to go by and it's due to my own unfamiliarity with the tools, I'm sorry for that! ^^'. 

> In the log that you have attached you can see three command lines at the end, just before the errors. Can you try to execute one of them and see if it fails? If it does, try to figure out what part of the command line is causing this (I don't see anything obvious).

Unfortunately I'm hitting the same wall as you and, other than what I mentioned above, I really don't see anything obvious either which is just as frustrating on your end as it is on mine 🫠

**Comment by wroyca** [comment]
> > I really don't see anything obvious either [...]
> 
> But did you try to run the failing command from the log?

My bad! Forgot to mention; yes, I ran the failed command back when I was reading the msdn doc in parallel to figure things out. It been two week since then so the memory is fuzzy, but I will try to look more into it this weekend and see if I can provide more information - unfortunately I am not currently available to do so until then

**Comment by wroyca** [comment]
@Klaim Can you assign me to this issue? I will try to solve it next week \:)

**Comment by wroyca** [comment]
Fixed with 01665e5501954961b6f72ce50d2fc3cbfadcddf6
